Source/GSThemeDrawing.m

```diff
@@ -3531,8 +3534,12 @@ - (void) drawOutlineViewRow: (NSInteger)rowIndex
   NSInteger i;
   id dataSource = [outlineView dataSource];
   NSTableColumn *outlineTableColumn = [outlineView outlineTableColumn];
+  GSKeyValueBinding *theBinding = nil;
 
-  if (dataSource == nil)
+  theBinding = [GSKeyValueBinding getBinding: NSContentBinding 
```

Comment:
We should not need to access bindings from the theme drawing code. If we really have to, we should rearrange the code to get better input parameters.

Reply:
I will take a look at this.

Reply:
I am not certain this is even needed.

Reply:
The reason this is checked here is only because we need to make sure there is some data source that is available whether it is a traditional one or a binding.   I can add something to check this so that it doesn't need to be checked locally.  Either adding a parameter or adding a private ivar/accessor.

Reply:
I have moved the check into [NSOutlineView drawRow:clipRect] so that it is not theme specific.  I have checked both view based on non-view based outline view tests and they work fine.